flink/src/test/java/org/apache/iceberg/flink/data/TestHelpers.java

```diff
@@ -0,0 +1,198 @@
+
+  public static void assertRowData(Type type, LogicalType rowType, Record expectedRecord, RowData actualRowData) {
+    if (expectedRecord == null && actualRowData == null) {
+      return;
+    }
+
+    Assert.assertTrue("expected Record and actual RowData should be both null or not null",
+        expectedRecord != null && actualRowData != null);
+
+    List<Type> types = Lists.newArrayList();
+    for (Types.NestedField field : type.asStructType().fields()) {
+      types.add(field.type());
+    }
+
+    for (int i = 0; i < types.size(); i += 1) {
+      if (expectedRecord.get(i) == null) {
+        Assert.assertTrue(actualRowData.isNullAt(i));
+        continue;
+      }
+
+      Object expected = expectedRecord.get(i);
+      LogicalType logicalType = ((RowType) rowType).getTypeAt(i);
+
+      final int fieldPos = i;
+      assertEquals(types.get(i), logicalType, expected,
+          () -> RowData.createFieldGetter(logicalType, fieldPos).getFieldOrNull(actualRowData));
+    }
+  }
+
+  private static void assertEquals(Type type, LogicalType logicalType, Object expected, Supplier<Object> supplier) {
+    switch (type.typeId()) {
+      case BOOLEAN:
+        Assert.assertEquals("boolean value should be equal", expected, supplier.get());
+        break;
+      case INTEGER:
+        Assert.assertEquals("int value should be equal", expected, supplier.get());
+        break;
+      case LONG:
+        Assert.assertEquals("long value should be equal", expected, supplier.get());
+        break;
+      case FLOAT:
+        Assert.assertEquals("float value should be equal", expected, supplier.get());
+        break;
+      case DOUBLE:
+        Assert.assertEquals("double value should be equal", expected, supplier.get());
+        break;
+      case STRING:
+        Assert.assertTrue("Should expect a CharSequence", expected instanceof CharSequence);
+        Assert.assertEquals("string should be equal",
+            String.valueOf(expected), supplier.get().toString());
+        break;
+      case DATE:
+        Assert.assertTrue("Should expect a Date", expected instanceof LocalDate);
+        LocalDate date = ChronoUnit.DAYS.addTo(EPOCH_DAY, (int) supplier.get());
+        Assert.assertEquals("date should be equal", expected, date);
+        break;
+      case TIME:
+        Assert.assertTrue("Should expect a LocalTime", expected instanceof LocalTime);
+        int milliseconds = (int) (((LocalTime) expected).toNanoOfDay() / 1000_000);
+        Assert.assertEquals("time millis should be equal", milliseconds, supplier.get());
+        break;
+      case TIMESTAMP:
+        if (((Types.TimestampType) type).shouldAdjustToUTC()) {
+          Assert.assertTrue("Should expect a OffsetDataTime", expected instanceof OffsetDateTime);
+          OffsetDateTime ts = (OffsetDateTime) expected;
+          Assert.assertEquals("OffsetDataTime should be equal", ts.toLocalDateTime(),
+              ((TimestampData) supplier.get()).toLocalDateTime());
```

Comment:
Why does this convert to `LocalDateTime` before validating? Can `TimestampData` be converted to `OffsetDateTime`?

Reply:
There is no direct method to convert `TimestampData` to `OffsetDateTime`.